### PR TITLE
[AMDGPU][True16][MC] VOP2 update instructions with fake16 format

### PR DESCRIPTION
**Pull request by broxigarchen**
Some old "t16" VOP2 instructions are actually in fake16 format. Correct and update test file

### CODE REVIEW

**Comment by broxigarchen** [comment]
failed test `tools/llvm-tli-checker/ps4-tli-check.yaml` seems unrelated with this patch

File: llvm/lib/Target/AMDGPU/VOP2Instructions.td [hunk 1]
```diff
@@ -1675,14 +1700,16 @@ multiclass VOP3Only_Realtriple_t16_and_fake16_gfx11_gfx12<bits<10> op, string as
 multiclass VOP3beOnly_Realtriple_gfx11_gfx12<bits<10> op> :
   VOP3beOnly_Realtriple<GFX11Gen, op>, VOP3beOnly_Realtriple<GFX12Gen, op>;
 
-multiclass VOP2Only_Real_MADK_with_name_gfx11_gfx12<bits<6> op, string asmName,
+multiclass VOP2Only_Real_MADK_t16_gfx11_gfx12<bits<6> op, string asmName,
                                                     string opName = NAME> :
```
Comment (kosarev) [inline llvm/lib/Target/AMDGPU/VOP2Instructions.td:1704]: Needs to be indented too.

File: llvm/lib/Target/AMDGPU/SIInstrInfo.cpp [hunk 2]
```diff
@@ -3852,19 +3852,22 @@ MachineInstr *SIInstrInfo::convertToThreeAddress(MachineInstr &MI,
     return MIB;
   }
 
+// FIXME-TRUE16. assert should be enabled after V_FMAC_F16_t16 is enabled
+#if 0
   assert(Opc != AMDGPU::V_FMAC_F16_t16_e32 &&
```
[the remark this answers is not in the document]
Comment (broxigarchen) [reply llvm/lib/Target/AMDGPU/SIInstrInfo.cpp:3857]: That's right! I forgot to rollback this part when I am testing another issue. Correct this

File: llvm/lib/Target/AMDGPU/VOP2Instructions.td [hunk 3]
```diff
@@ -988,16 +998,17 @@ let FPDPRounding = 1, isReMaterializable = 1, FixedSize = 1 in {
 let SubtargetPredicate = isGFX10Plus, True16Predicate = NotHasTrue16BitInsts in {
 def V_FMAMK_F16 : VOP2_Pseudo <"v_fmamk_f16", VOP_MADMK_F16, [], "">;
 }
-let SubtargetPredicate = HasTrue16BitInsts in {
-def V_FMAMK_F16_t16 : VOP2_Pseudo <"v_fmamk_f16_t16", VOP_MADMK_F16_t16, [], "">;
+let True16Predicate = UseFakeTrue16Insts in {
+ def V_FMAMK_F16_fake16 : VOP2_Pseudo <"v_fmamk_f16_fake16", VOP_MADMK_F16_fake16, [], "">;
```
[the remark this answers is not in the document]
Comment (broxigarchen) [reply llvm/lib/Target/AMDGPU/VOP2Instructions.td]: I'll keep two space identation. done

File: llvm/lib/Target/AMDGPU/VOP2Instructions.td [hunk 4]
```diff
@@ -988,16 +998,17 @@ let FPDPRounding = 1, isReMaterializable = 1, FixedSize = 1 in {
...
 }
 
+
```
[the remark this answers is not in the document]
Comment (broxigarchen) [reply llvm/lib/Target/AMDGPU/VOP2Instructions.td]: done

File: llvm/lib/Target/AMDGPU/VOP2Instructions.td (hunk 1 above)
Comment (broxigarchen) [reply llvm/lib/Target/AMDGPU/VOP2Instructions.td]: done

**Review by kosarev — APPROVED** [review]
LGTM.

**Review by Sisyph — APPROVED** [review]
LGTM

**Pull request merged**